### PR TITLE
feat: clean up deployment artifacts

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Various deployment artifact cleanups:
1. Removed `mainnet-trial` and `mainnet-trial-2` artifacts.
2. Rewrote `generate-markdown` in TypeScript.
3. Modified `generate-markdown` so that it generates READMEs for each deployment instead of generating one massive README.
3. Added `AddressDictator` and `ChugSplashDictator` to the list of hidden addresses.

### CODE REVIEW

**Review by maurelian — CHANGES_REQUESTED** [review]

File: packages/contracts/deployments/goerli/README.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+<img width="506px" height="0px" />
+<p><small>Address</small></p>
+</th>
+</tr>
+<tr>
+<td>
+OVM_L2ToL1MessagePasser
+</td>
+<td align="center">
+<a href="undefined/address/0x4200000000000000000000000000000000000000">
```
[the remark this answers is not in the document]
Comment (smartcontracts) [reply packages/contracts/deployments/goerli/README.md:127]: Lol, whoops. Good catch.

File: packages/contracts/deployments/goerli/README.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
+# Optimistic Goerli (internal devnet)
+## Notice
+Optimistic Goerli is an internal Optimism development network. You're probably looking for [Optimistic Kovan](../kovan), the public Optimistic Ethereum testnet.
```
[the remark this answers is not in the document]
Comment (smartcontracts) [reply packages/contracts/deployments/goerli/README.md:3]: Great point.

**Review by maurelian — APPROVED** [review]
LGTM

File: packages/contracts/scripts/generate-markdown.ts [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+ * @returns Nicely formatted markdown-compatible table as a string.
+ */
+const buildContractsTable = (
+  contracts: ContractInfo[],
+  explorer?: string
+): string => {
+  // Being very verbose within this function to make it clear what's going on.
+  // We use HTML instead of markdown so we can get a table that displays well on GitHub.
+  // GitHub READMEs are 1012px wide. Adding a 506px image to each table header is a hack that
+  // allows us to create a table where each column is 1/2 the full README width.
```
Comment (annieke) [inline packages/contracts/scripts/generate-markdown.ts:98]: nice

**Review by annieke — APPROVED** [review]
checked the new deployment md files and they look good, `generate-markdown.ts` is also v easy to read so lgtm!

**Pull request merged**